### PR TITLE
Fixes #29642 - drop sqlite support

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

File: lib/tasks/backup.rake [hunk 1]
```diff
@@ -45,8 +41,6 @@ namespace :db do
     case config['adapter']
     when 'postgresql'
       postgres_dump(backup_name, config, tables)
-    when 'sqlite3'
-      sqlite_dump(backup_name, config)
     else
       puts 'Your database is not supported by Foreman.' and exit(1)
     end
```
Comment (ekohl) [inline lib/tasks/backup.rake:46]: It feels like we can do this check at line 23 and error out early, but I'm ok with keeping this patch minimal.

**Review by ekohl — COMMENTED** [review]
How about:

https://github.com/theforeman/foreman/blob/19715ec794e5dca5aed738f9b9779c0daaf97250/test/models/compute_resource_test.rb#L211
https://github.com/theforeman/foreman/blob/19715ec794e5dca5aed738f9b9779c0daaf97250/test/controllers/subnets_controller_test.rb#L28
https://github.com/theforeman/foreman/blob/19715ec794e5dca5aed738f9b9779c0daaf97250/test/controllers/subnets_controller_test.rb#L39

**Comment by tbrisker** [comment]
Removed the explicit sqlite checks in tests, there are probably other places that can be cleaned up but that can be done in separate prs.

**Review by ekohl — DISMISSED** [review]

File: config/application.rb [hunk 2]
```diff
@@ -134,9 +134,6 @@ class Application < Rails::Application
     # Configure sensitive parameters which will be filtered from the log file.
     config.filter_parameters += [:password, :account_password, :facts, :root_pass, :value, :report, :password_confirmation, :secret]
 
-    # Since Rails 6 sqlite db must use integer representation for booleans
-    config.active_record.sqlite3.represent_boolean_as_integer = true
```
Comment (mmoll) [inline config/application.rb:138]: `config/as_deprecation_whitelist.yaml` can also get adjusted then.

Comment (tbrisker) [reply config/application.rb:138]: good point, fixed.

**Review by ezr-ondrej — APPROVED** [review]
LGTM :+1:

**Review by ekohl — APPROVED** [review]
@mmoll last look before we merge it? Would also be good to get ACKs on the packaging PRs before we merge it.

**Review by mmoll — APPROVED** [review]
looks good to me

**Comment by tbrisker** [comment]
deb: https://github.com/theforeman/foreman-packaging/pull/5131
rpm: https://github.com/theforeman/foreman-packaging/pull/5132

**Comment by tbrisker** [comment]
@mmoll @ekohl @ezr-ondrej who wants to push the button? Packaging PRS are ready and blocked on this being merged

**Comment by ekohl** [comment]
Thanks! Once the source tarball release job is done, the packaging PRs should be kicked off again.

**Pull request merged**